From 61564e7761e38e5ec55e7939ccd6a276c2c55d11 Mon Sep 17 00:00:00 2001
From: Miles Granger <miles59923@gmail.com>
Date: Sat, 4 Jan 2025 07:10:13 +0100
Subject: [PATCH] Remove experimental codecs in default builds (#197)

Removes experimental blosc2, igzip, ideflate, and izlib from default builds
Still available by building from source w/ features enabled.

cbouss: adapted for 2.9.1
---
 .github/workflows/CI.yml | 79 ++--------------------------------------
 Cargo.lock               |  2 +-
 Cargo.toml               |  6 ++-
 README.md                |  4 ++
 benchmarks/test_bench.py |  6 ++-
 pyproject.toml           |  2 +-
 tests/test_blosc2.py     | 13 +++++--
 7 files changed, 28 insertions(+), 84 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index cd78f573..0ed6e88a 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -14,7 +14,9 @@ name = "cramjam"
 crate-type = ["cdylib"]
 
 [features]
-default                  = ["extension-module", "snappy", "lz4", "bzip2", "brotli", "xz", "zstd", "gzip", "zlib", "deflate", "blosc2", "igzip", "ideflate", "izlib"]
+default                  = ["extension-module", "snappy", "lz4", "bzip2", "brotli", "xz", "zstd", "gzip", "zlib", "deflate"]
+experimental             = ["blosc2", "igzip", "ideflate", "izlib"]
+
 extension-module         = ["pyo3/extension-module"]
 generate-import-lib      = ["pyo3/generate-import-lib"]  # needed for Windows PyPy builds
 
diff --git a/tests/test_blosc2.py b/tests/test_blosc2.py
index c247b61d..7d3cf874 100644
--- a/tests/test_blosc2.py
+++ b/tests/test_blosc2.py
@@ -10,7 +10,12 @@
 except ImportError:
     pytest.skip("experimental module not built", allow_module_level=True)
 else:
-    blosc2 = experimental.blosc2
+    if hasattr(experimental, "blosc2"):
+        blosc2 = experimental.blosc2
+    else:
+        pytest.skip(
+            "experimental module doesn't contain blosc2", allow_module_level=True
+        )
 
 
 settings.register_profile("local", max_examples=10)
@@ -25,7 +30,7 @@
 def variants(e):
     for attr in dir(e):
         # TODO: LastCodec, LastFilter, LastRegisteredCodec/Filter not supported
-        if not attr.startswith('_') and not attr.lower().startswith('last'):
+        if not attr.startswith("_") and not attr.lower().startswith("last"):
             yield getattr(e, attr)
 
 
@@ -47,7 +52,9 @@ def test_roundtrip_chunk_into(data, codec, filter, clevel):
     kwargs = dict(clevel=clevel, filter=filter, codec=codec)
     nbytes_compressed = len(blosc2.compress_chunk(data, **kwargs))
 
-    compressed = np.empty(blosc2.max_compressed_len(len(data.tobytes())), dtype=np.uint8)
+    compressed = np.empty(
+        blosc2.max_compressed_len(len(data.tobytes())), dtype=np.uint8
+    )
     nbytes = blosc2.compress_chunk_into(data, compressed, **kwargs)
 
     decompressed = np.empty(len(data.tobytes()) * 2, dtype=np.uint8)
